portal-api/src/v1/validation/fields/currency-not-the-same-as-supply-contract-rules/currency.js

```diff
@@ -1,11 +1,12 @@
-const { hasValue } = require('../../../../utils/string');
 const { orderNumber } = require('../../../../utils/error-list-order-number');
 const { currencyIsDisabled } = require('../currency');
 
+// TODO - unit test
+
 module.exports = (facility, errorList) => {
   const newErrorList = { ...errorList };
 
-  if (!hasValue(facility.currency)) {
+  if (!facility.currency?.id) {
```

Comment:
This validation was causing issues because `hasValue` was returning `true` for an empty object, which means no value has been selected/mapped.

Reply:
I wonder if we need an isObjectEmpty function as a temp fix?

The return of empty objects is widespread and can cause a host of errors (noticeable and silent)... but thats another day's discussion. I'm not even sure where we would begin, but if we moved to an easily maintainable existing library, like zod or joi, I reckon around 90% of validation code would be gutted.

Reply:
Completely agree - I did think about creating an `isEmptyObject` type helper but this should be thought about another time.

Reply:
We have lodash installed -- Could we use `_.isEmpty(facility.currency);` maybe?

Reply:
I don't see a need to use lodash here, `facility.currency?.id` does the job.

It's also a failsafe to specifically check for the ID - probably edge case, butt we don't want this to error if the object contains properties other than an ID